File: swagger-client/micronaut/src/main/java/io/swagger/model/MailBlockRspamd.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import io.micronaut.core.annotation.Introspected;
import io.micronaut.validation.Validated;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * A block pattern sourced from the rspamd database.  Represents a sender whose recent messages contained suspicious subject lines (e.g. containing relay/proxy strings) repeated more than 4 times in the last 3 days. The &#x60;from&#x60; address can be passed to &#x60;POST /mail/blocks/delete&#x60; to delist it.
 */
@Schema(description = "A block pattern sourced from the rspamd database.  Represents a sender whose recent messages contained suspicious subject lines (e.g. containing relay/proxy strings) repeated more than 4 times in the last 3 days. The `from` address can be passed to `POST /mail/blocks/delete` to delist it.")
@Validated
@Introspected

public class MailBlockRspamd   {
  @JsonProperty("from")
  private String from = null;

  @JsonProperty("subject")
  private String subject = null;

  public MailBlockRspamd from(String from) {
    this.from = from;
    return this;
  }

  /**
   * The sender email address.  Pass this value as `email` to `POST /mail/blocks/delete` to delist the sender.
   * @return from
  **/
  @Schema(example = "user@domain.com", required = true, description = "The sender email address.  Pass this value as `email` to `POST /mail/blocks/delete` to delist the sender.")
  @NotNull

  public String getFrom() {
    return from;
  }

  public void setFrom(String from) {
    this.from = from;
  }

  public MailBlockRspamd subject(String subject) {
    this.subject = subject;
    return this;
  }

  /**
   * The suspicious subject pattern that triggered the block.
   * @return subject
  **/
  @Schema(example = "Test email", required = true, description = "The suspicious subject pattern that triggered the block.")
  @NotNull

  public String getSubject() {
    return subject;
  }

  public void setSubject(String subject) {
    this.subject = subject;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    MailBlockRspamd mailBlockRspamd = (MailBlockRspamd) o;
    return Objects.equals(this.from, mailBlockRspamd.from) &&
        Objects.equals(this.subject, mailBlockRspamd.subject);
  }

  @Override
  public int hashCode() {
    return Objects.hash(from, subject);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class MailBlockRspamd {\n");
    
    sb.append("    from: ").append(toIndentedString(from)).append("\n");
    sb.append("    subject: ").append(toIndentedString(subject)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: swagger-client/dart/lib/model/mail_block_rspamd.dart
part of swagger.api;

class MailBlockRspamd {
  /* The sender email address.  Pass this value as `email` to `POST /mail/blocks/delete` to delist the sender. */
  String from = null;
/* The suspicious subject pattern that triggered the block. */
  String subject = null;

  MailBlockRspamd();

  @override
  String toString() {
    return 'MailBlockRspamd[from=$from, subject=$subject, ]';
  }

  MailBlockRspamd.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    from = json['from'];
    subject = json['subject'];
  }

  Map<String, dynamic> toJson() {
    return {
      'from': from,
      'subject': subject
     };
  }

  static List<MailBlockRspamd> listFromJson(List<dynamic> json) {
    return json == null ? new List<MailBlockRspamd>() : json.map((value) => new MailBlockRspamd.fromJson(value)).toList();
  }

  static Map<String, MailBlockRspamd> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, MailBlockRspamd>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new MailBlockRspamd.fromJson(value));
    }
    return map;
  }
}
